treeherder/services/signatures_remover.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from typing import List
+from taskcluster import exceptions
+
+from treeherder.perf.models import PerformanceDatum, PerformanceSignature
+from treeherder.services.taskcluster import TaskclusterModel
+from treeherder.services.max_runtime import MaxRuntime
+
+logger = logging.getLogger(__name__)
+
+RECEIVER_TEAM_EMAIL = "perftest-alerts@mozilla.com"
+RECEIVER_EMAIL = "dhunt@mozilla.com"
+
+
+class PublicSignaturesRemover:
+    """
+    This class handles the removal of signatures which are (no longer)
+    associated with a job and sends email notifications to the entire team.
+    """
+
+    table_description = """__Summary of recently deleted Performance Signatures__
+> The signatures below were deleted due to the fact that they were no longer associated with a job.
+---
+    """
+
+    table_headers = """
+| Repository | Framework | Platform | Suite | Application |
+| :---: | :---: | :---: | :---: | :---: |
+    """
+
+    def __init__(
+        self,
+        timer: MaxRuntime,
+        taskcluster_model: TaskclusterModel,
+    ):
+        self.tc_model = taskcluster_model
+        self._subject = "Summary of deleted Performance Signatures"
+        self._content = None
+        self._nr_of_rows_allowed = 50
+        self._nr_of_emails_allowed = 10
+
+        self.timer = timer
+
+    def remove(self):
+        emails_sent = 0
+
+        rows_left = self._nr_of_rows_allowed
+        chunk_of_signatures = []
+
+        for performance_signature in PerformanceSignature.objects.all():
+            self.timer.quit_on_timeout()
+
+            if emails_sent != self._nr_of_emails_allowed and (
+                not PerformanceDatum.objects.filter(
+                    repository_id=performance_signature.repository_id,  # leverages (repository, signature) compound index
+                    signature_id=performance_signature.id,
+                ).exists()
+            ):
+                rows_left -= 1
+                chunk_of_signatures.append(performance_signature)
+
+                if rows_left == 0:
+                    self._send_notification(chunk_of_signatures)
+
+                    emails_sent += 1
+                    chunk_of_signatures = []
+                    rows_left = self._nr_of_rows_allowed
+
+            if emails_sent == self._nr_of_emails_allowed:
+                break
+
+        if emails_sent != self._nr_of_emails_allowed and chunk_of_signatures != []:
+            self._send_notification(chunk_of_signatures)
```

Comment:
I would get `signature.delete()` out of `_send_notification()` and rename `def remove()` to `def remove_in_chunks()` like this:
```
...
        if emails_sent != self._nr_of_emails_allowed and chunk_of_signatures != []:
            self._send_notification(chunk_of_signatures)
            self._delete(chunk_of_signatures)

...

    def _delete(self, chunk):
        for signature in chunk:
            signature.delete()

    def _send_notification(self, chunk_of_signatures):
        # check if Taskcluster Notify Service is up
        try:
            self._ping_email_service()
        except exceptions.TaskclusterConnectionError:
            logger.info("Taskcluster Notify Service is not available.")
        # extract the proprieties of interest from signatures in a list of dictionaries
        email_content = self.__extract_properties(chunk_of_signatures)
        # delete signatures
        for signature in chunk_of_signatures:
            signature.delete()
        # send email to the team with summary of signatures just deleted
        self._send_email(RECEIVER_TEAM_EMAIL, email_content)
        self._send_email(RECEIVER_EMAIL, email_content)
```